files/build_templates/docker_image_ctl.j2

```diff
@@ -44,12 +44,19 @@ function updateSyslogConf()
     fi
     CONTAINER_NAME="$DOCKERNAME"
     TMP_FILE="/tmp/rsyslog.$CONTAINER_NAME.conf"
+    failed="false"
     {%- if docker_container_name == "database" %}
     python -c "import jinja2, os; paths=['/usr/share/sonic/templates']; loader = jinja2.FileSystemLoader(paths); env = jinja2.Environment(loader=loader, trim_blocks=True); template_file='/usr/share/sonic/templates/rsyslog-container.conf.j2'; template = env.get_template(os.path.basename(template_file)); data=template.render({\"target_ip\":\"$TARGET_IP\",\"container_name\":\"$CONTAINER_NAME\"}); print(data)" > $TMP_FILE
     {%- else %}
     sonic-cfggen -d -t /usr/share/sonic/templates/rsyslog-container.conf.j2 -a "{\"target_ip\": \"$TARGET_IP\", \"container_name\": \"$CONTAINER_NAME\" }"  > $TMP_FILE
+    if [ $? -ne 0 ]; then
+      echo "Error: Execute sonic-cfggen failed."
+      failed="true"
+    fi
     {%- endif %}
-    docker cp $TMP_FILE ${DOCKERNAME}:/etc/rsyslog.conf
+    if [ "$failed" == "false" ]; then
+      docker cp $TMP_FILE ${DOCKERNAME}:/etc/rsyslog.conf
+    fi
```

Comment:
There is a risk that teamd container started without proper rsyslog.conf file. Then probably the log of teamd container can't be received by the host rsyslog service. The consequence is that teamd syslog could be missing.

Reply:
Thanks for the reminder! Will discuss and improve again.